lib/index.js

```diff
@@ -974,3 +984,23 @@ exports.shallow = function (source) {
 
     return target;
 };
+
+
+exports.promiseWrap = function (bind, method, args, transform) {
+
+    return new Promise((resolve, reject) => {
+
+        const callback = function (error, result) {
+
+            if (Array.isArray(transform)) {
+                result = exports.transform(Array.from(arguments), transform);
+                error instanceof Error ? reject(result) : resolve(result);
```

Comment:
in the case of catbox, it can return an error and also a result when generating a new value fails but it returns a stale value https://github.com/hapijs/catbox/blob/master/test/policy.js#L1108